homeassistant/components/xiaomi_ble/sensor.py

```diff
@@ -163,6 +165,45 @@ def sensor_update_to_bluetooth_data_update(
     )
 
 
+def process_service_info(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    data: XiaomiBluetoothDeviceData,
+    service_info: BluetoothServiceInfoBleak,
+) -> PassiveBluetoothDataUpdate:
+    """Process a BluetoothServiceInfoBleak, running side effects and returning sensor data."""
+    update = data.update(service_info)
+
+    # If device isn't pending we know it has seen at least one broadcast with a payload
+    # If that payload was encrypted and the bindkey was not verified then we need to reauth
+    if (
+        not data.pending
+        and data.encryption_scheme != EncryptionScheme.NONE
+        and not data.bindkey_verified
+    ):
+        flow_context = {
+            "source": config_entries.SOURCE_REAUTH,
+            "entry_id": entry.entry_id,
+            "title_placeholders": {"name": entry.title},
+            "unique_id": entry.unique_id,
+            "device": data,
+        }
+
+        for flow in hass.config_entries.flow.async_progress_by_handler(DOMAIN):
+            if flow["context"] == flow_context:
+                break
+        else:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
```

Comment:
Maybe we should make a config entries helper to start a re-auth flow with custom context items? It would be nice if we didn't have to touch the standard items or repeat the logic around flows when starting a re-auth flow.

Reply:
I was going to come back with a seperate PR for that because i felt the same. Was considering just adding a `context=None` optional to the existing one, that was merged in it if was present.

Reply:
Sounds good!